packages/nimble/inst/tests/test_utils.R

```diff
@@ -51,22 +51,28 @@ indexNames <- function(x) {
     lapply(x, function(z) {z$name <- paste(i, z$name); i <<- i + 1; z})
 }
 
-test_coreRfeature <- function(input, verbose = TRUE, dirName = NULL) { ## a lot like test_math but a bit more flexible
+
+test_coreRfeature <- function(input, verbose = TRUE, dirName = NULL) {
+    test_that(input$name, {
+        test_coreRfeature_internal(input, verbose, dirName)
+    })
+}
+
+test_coreRfeature_internal <- function(input, verbose = TRUE, dirName = NULL) { ## a lot like test_math but a bit more flexible
   if(verbose) cat("### Testing", input$name, "###\n")
   runFun <- gen_runFunCore(input)
   nfR <- nimbleFunction(run = runFun)
-  nfC <- try(compileNimble(nfR, dirName = dirName))
+  ## This try is safe because failure is caught by expect_equal below
+  nfC <- try(compileNimble(nfR, dirName = dirName), silent = TRUE)
```

Comment:
This looks reasonable. I'm not sure it's appropriate here, but I've also found `expec_error` helpful in `test_size()`:
```r
if(failureExpected) {
  expect_error(compileNimble(...))
} else {
  nf <- compileNimble(...)
}
```